### PR TITLE
z0m lai dependence

### PR DESCRIPTION
**Pull request by barlage**

### CODE REVIEW

**Comment by barlage** [comment]
@wzzheng90 @RongqianYang-NOAA @HelinWei-NOAA This code is not answer changing but has the ability to set an option in noahmpdrv for z0m dependence on LAI using the method from CLM. Check it out and let me know if you have any questions or suggestions. There are also commits for documenting z0h subroutine.

File: physics/module_sf_noahmplsm.F90 [hunk 1]
```diff
@@ -5487,143 +5521,186 @@ end subroutine sfcdif3
...
+
+      z0m_out = z0mg
+     
+      if (opt_trs == z0heqz0m) then
+
+        z0h_out = z0m_out
+
+      elseif (opt_trs == chen09) then
+
+        czil = 10.0 ** (- 0.4 * parameters%hvt)
```
Comment (HelinWei-NOAA) [inline physics/module_sf_noahmplsm.F90:5646]: Do we want to use the fixed czil for the bare soil?

**Comment by HelinWei-NOAA** [comment]
@wzzheng90 @RongqianYang-NOAA Can you review the change?

**Comment by RongqianYang-NOAA** [comment]
> @wzzheng90 @RongqianYang-NOAA Can you review the change?

Yes.

**Comment by wzzheng90** [comment]
> @wzzheng90 @RongqianYang-NOAA Can you review the change?

Sure.

**Pull request merged**